simulator/infra/trap_types.h

```diff
@@ -0,0 +1,21 @@
+/* trap_types.h - Trap types for MIPS and RISC-V
```

Comment:
That file should be moved to.... FuncSim directory, I believe.
In my opinion, `infra` should contain generic things, unrelated to system and ISA stuff.

Reply:
Yes, but it's used by `mips_instr` and `riscv_instr`. If we had a base ISA-independent class for instruction, it would fit well in it. Also, it can be moved to trap implementation.

Reply:
As I see now, it should be moved to trap class. At this moment, should it be moved to FuncSim, or I can leave it as is?

Reply:
I'd keep "everything related to functional state" in func_sim: RF, system calls, trap handler etc.. Later we'll use better name, I believe.